src/main/java/org/takes/http/BkBasic.java

```diff
@@ -70,17 +77,34 @@ public BkBasic(final Take tks) {
         this.take = tks;
     }
 
+//      @todo #519:30min Need to support Keep-Alive header here and take
+//       into account specified timeout and max requests parameters.
+//       See http://tools.ietf.org/id/draft-thomson-hybi-http-timeout-01.html
+//       for details. Also need to remove input.available <= 0 condition
+//       - this is done in order to prevent failure of some tests,
+//       should be replaced with proper Keep-Alive header in tests
     @Override
+    @SuppressWarnings({
+        "PMD.EmptyCatchBlock",
+        "PMD.AvoidInstantiatingObjectsInLoops"})
     public void accept(final Socket socket) throws IOException {
+        socket.setSoTimeout(BkBasic.TIMEOUT);
         final InputStream input = socket.getInputStream();
         try {
-            this.print(
-                BkBasic.addSocketHeaders(
-                    new RqLive(input),
-                    socket
-                ),
-                new BufferedOutputStream(socket.getOutputStream())
-            );
+            while (true) {
+                final Request request = new RqLive(input);
+                this.print(
+                    BkBasic.addSocketHeaders(
+                        request,
+                        socket
+                    ),
+                    new BufferedOutputStream(socket.getOutputStream())
+                );
+                if (input.available() <= 0) {
+                    break;
+                }
+            }
+        } catch (final SocketTimeoutException exc) {
```

Comment:
@krzyk we are reusing the same connection for a limited time - if time is over we just need to close the socket. No need to do anything specific on timeout, at the same time it's not an error(no need to throw it on upper level)